helix-core/src/main/java/org/apache/helix/manager/zk/ZkBaseDataAccessor.java

```diff
@@ -142,22 +142,22 @@ public ZkBaseDataAccessor(RealmAwareZkClient zkClient) {
   }
 
   private ZkBaseDataAccessor(Builder<T> builder) {
-    switch (builder.realmMode) {
+    switch (builder.getRealmMode()) {
```

Comment:
The following logic looks generic enough to put in GenericZkHelixApiBuilder. And this ZkBaseDataAccessor constructor can directly get a zkclient instance as the input.